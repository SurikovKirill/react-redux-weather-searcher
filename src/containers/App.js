import React, { Component } from 'react';
import { connect } from 'react-redux';
import { initWeather } from '../actions/city';
import './App.css';
import CityContainer from './CityContainer';
import Header from '../components/Header';

class App extends Component {
  constructor(props) {
      super(props);
      props.initWeather();
  }

  geolocationWeather = () => {
      this.props.initWeather();
  };

  render() {
      const { isLoading, weather, error } = this.props;
      return (
          <div className='App'>
              <Header onUpdate={this.geolocationWeather}/>
              <CityContainer weather={weather} isLoading={isLoading} error={error}/>
          </div>
      );
  }
}

const mapStateToProps = ({ cityReducer: { isLoading, error, weather } }) => {
  return {
      isLoading,
      error,
      weather,
  };
};
const mapDispatchToProps = {
  initWeather: initWeather,
};

export default connect(mapStateToProps, mapDispatchToProps)(App);