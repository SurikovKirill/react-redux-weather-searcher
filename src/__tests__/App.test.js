import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../containers/App';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('App container', () => {

    test('matches the snapshot when local weather is loaded and favorites cities are loaded', () => {
        const mockGeolocation = {
            getCurrentPosition: jest.fn()
                .mockImplementation((success) => Promise.resolve(success({
                    coords: {
                        latitude: 10,
                        longitude: 10
                    }
                })))
        };
        navigator.geolocation = mockGeolocation;
        const city = {
            isLoading: false,
            error: false,
            weather: {
                name: 'Moscow',
                img: 'i',
                temperature: 't',
                wind: 'w',
                cloudiness: 'c',
                pressure: 'p',
                humidity: 'h',
                location: 'l'
            }
        };
        const cities = [{
            name: 'Moscow',
            img: 'i1',
            temperature: 't1',
            wind: 'w1',
            cloudiness: 'c1',
            pressure: 'p1',
            humidity: 'h1',
            location: 'l1'
        }];
        const store = mockStore({
            cityReducer: city,
            favouritesReducer: {
                cities: cities,
                errors: [],
                isLoading: []
            }
        });
        const tree = renderer.create(
            <Provider store={store}>
                <App />
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

});