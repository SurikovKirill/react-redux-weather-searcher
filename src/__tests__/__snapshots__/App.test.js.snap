// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container matches the snapshot when local weather is loaded and favorites cities are loaded 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1
      className="App-title"
    >
      Погода здесь
    </h1>
    <button
      className="Refresh-Location-Button"
      onClick={[Function]}
    >
      Обновить геолокацию
    </button>
  </header>
  <div
    className="Main"
  >
    <div
      className="currentCity"
    >
      <h1>
        Moscow
      </h1>
      <div
        className="additionalInfo"
      >
        <img
          alt="Иконка погоды"
          className="weatherImage"
          src="i"
        />
        <h1>
          t
        </h1>
      </div>
    </div>
    <div
      className="WeatherInfo"
    >
      <div
        className="Info"
      >
        <h3>
          Ветер
        </h3>
        <h4>
          w
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Облачность
        </h3>
        <h4>
          c
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Давление
        </h3>
        <h4>
          p
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Влажность
        </h3>
        <h4>
          h
        </h4>
      </div>
      <div
        className="Info"
      >
        <h3>
          Координаты
        </h3>
        <h4>
          l
        </h4>
      </div>
    </div>
  </div>
  <div
    className="search"
  >
    <h1>
      Избранное
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="searchInput"
        name="cityName"
        placeholder="Добавить новый город"
        required="required"
      />
      <button
        className="searchButton"
        type="submit"
      />
    </form>
  </div>
  <div
    className="FavList"
  >
    <div
      className="FavouriteCity"
    >
      <div
        className="favCityHeader"
      >
        <h1>
          Moscow
        </h1>
        <img
          alt="Иконка погоды"
          src="i1"
        />
        <h1>
          t1
        </h1>
        <button
          className="deleteButton"
          onClick={[Function]}
        >
          X
        </button>
      </div>
      <div
        className="WeatherInfo"
      >
        <div
          className="Info"
        >
          <h3>
            Ветер
          </h3>
          <h4>
            w1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Облачность
          </h3>
          <h4>
            c1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Давление
          </h3>
          <h4>
            p1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Влажность
          </h3>
          <h4>
            h1
          </h4>
        </div>
        <div
          className="Info"
        >
          <h3>
            Координаты
          </h3>
          <h4>
            l1
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
